readme.nix

```diff
@@ -15,25 +15,37 @@ let
   docparse = naersk.buildPackage {
     root = ./docparse;
 
-    src = builtins.filterSource (
-      p: t:
-        let
-          p' = pkgs.lib.removePrefix (toString ./docparse + "/") p;
-        in
-        p' == "Cargo.lock" || p' == "Cargo.toml" || p' == "src" || p' == "src/main.rs"
-    ) ./docparse;
+    src = builtins.filterSource
+      (
+        p: t:
+          let
+            p' = pkgs.lib.removePrefix (toString ./docparse + "/") p;
+          in
+          p' == "Cargo.lock" || p' == "Cargo.toml" || p' == "src" || p' == "src/main.rs"
+      ) ./docparse;
   };
 
-in rec {
-  body = pkgs.runCommand "readme-body" {
-    buildInputs = [ docparse ];
-  } ''
-    cat ${./README.tpl.md} > $out
-    docparse ${./config.nix} >> gen
-    sed -e '/GEN_CONFIGURATION/{r gen' -e 'd}' -i $out
-  '';
+in
+rec {
+  body =
+    let
+      readme = builtins.readFile ./README.tpl.md;
+
+      params = builtins.readFile (
+        pkgs.runCommand "docparse"
+          { buildInputs = [ docparse ]; }
+          "docparse ${./config.nix} > $out"
+      );
+
+    in
+    pkgs.writeText "readme" (
+      builtins.replaceStrings
```

Comment:
The sed-based approach didn't work on MacOS:
```
; ./script/gen
Updating README
this derivation will be built:
  /nix/store/pfvinxyrj3x27wrmvlcw0zbzkwlfn2kn-readme-body.drv
building '/nix/store/pfvinxyrj3x27wrmvlcw0zbzkwlfn2kn-readme-body.drv'...
sed: couldn't open temporary file /nix/store/sedbfv96n: Operation not permitted
error: builder for '/nix/store/pfvinxyrj3x27wrmvlcw0zbzkwlfn2kn-readme-body.drv' failed with exit code 4;
       last 1 log lines:
       > sed: couldn't open temporary file /nix/store/sedbfv96n: Operation not permitted
       For full logs, run 'nix log /nix/store/pfvinxyrj3x27wrmvlcw0zbzkwlfn2kn-readme-body.drv'.
```
... so I figured that switching to an (almost) pure-Nix approach will be better.

Reply:
This has the unfortunate side effect of importing from derivation!

Reply:
Yeah, but that's just run locally, right? Naersk itself remains fully IFD-free 😇

Reply:
heh alright

Reply:
btw that sed error looks odd, I think it's just a matter of changing the `sed` slightly on macos